### ISSUE TITLE
Android - "Can't play this video" error with Videos on Azure Blob Storage

### ISSUE THREAD

**Issue by EdwardMcFlurry** [davidbritch/dotnet-maui-videoplayer#19]
Hello

I followed the previous issue regarding this error on Android however, mine is a little bit different in that the videos (.mp4 & .mov) were playing just fine when they were hosted along side my .Net Core MVC application.

I moved those videos to Azure Blob Storage without changing anything such as file names etc and yet the error pops up on Android and yet all the videos are loaded just fine on iOS and can play them.

Here is the strange thing, today I cloned this application and ran it just to confirm that it is working as expected on both iOS simulator and Android emulator and all was good. I changed the source of the Video to one of the Videos currently on Azure but it failed even on this project.

I can provide the links to those videos if anyone is willing to reproduce the issue with Android.

**Comment by EdwardMcFlurry** [comment]
@davidbritch please help

**Comment by AlexJinx** [comment]
I got the same error, how did you solve it

**Comment by EdwardMcFlurry** [comment]
@AlexJinx my issue is with video files using URI as the source, i don’t know your specific issue.

**Comment by AlexJinx** [comment]
> @AlexJinx我的问题是使用 URI 作为源的视频文件，但我不知道您的具体问题。

I solved the problem， in Platforms\Android\VideoProvider.cs file, ContentProvider Attribute value is com.companyname.projectname？

look this ：[ContentProvider(new string[] { "com.companyname.mauiclient" })]

**Comment by EdwardMcFlurry** [comment]
Hello @davidbritch please help

**Comment by EdwardMcFlurry** [comment]
> > @AlexJinx我的问题是使用 URI 作为源的视频文件，但我不知道您的具体问题。
> 
> I solved the problem， in Platforms\Android\VideoProvider.cs file, ContentProvider Attribute value is com.companyname.projectname？
> 
> look this ：[ContentProvider(new string[] { "com.companyname.mauiclient" })]

What exactly is the purpose of this file “VideoProvide”? And where is it referenced?

**Comment by EdwardMcFlurry** [comment]
@davidbritch I started a new demo project and went as far as having the Media View/Element  @jfversluis created and tested this one particular issue.

I’ve learned that Media View/Element doesn’t say “Can’t play this video” when the URI is invalid and also the same video currently hosted on Azure Blob Storage can play on Android.

All I wanted was for you to try to explain why there’s issue on your project and perhaps a workaround since Media View/Element is far from being finished.

**Comment by jfversluis** [comment]
What URL are you testing and what do you expect to happen if it works or if it doesn't work?

Because I would like to think that the MediaElement is close to being done for a first version and would love to fix this for you so that it will be even better.

**Comment by EdwardMcFlurry** [comment]
Hi @jfversluis 

Before this project was on even on Github, I had already begun my App whilst keeping an eye on different Teams working on a way to play videos in Maui including MauiToolkit, VLC etc and boom @davidbritch came with this project.

Initially I had uploaded Test videos alongside my ASP.NET Core MVC on some server in SA and used them as a Source property on this project and worked well on both Android & iOS.

I moved those videos as they were to Azure Blob Storage early December, changed the URI to point to the new Server and tested on iOS mostly as everything worked well but neglected to check if Android was still good. Then, all of a sudden this ISSUE popped on Android using the same videos I used before.

I asked about this ISSUE on Android with videos on Azure but failed to get response hence today I finally tried out Media View/Element as I’ve been following along since it dropped on YouTube.

I apologize for the long response but I’m expecting all the videos uploaded on Azure Blob Storage on play on Android as they currently do play on iOS 16.

This morning, I was able to play few of those video on Media View/Element on Android but one particular video that is 09:51 in length did not load/play or maybe I was impatient but I don’t know.

Again sorry 😞 for the long message.

**Comment by jfversluis** [comment]
Thanks for being extensive which is great! But without any technical details it will be hard to look at concrete things I can take action on to make this scenario work with our upcoming control for this.

If possible it would be very helpful to get links to the videos you are referring to or at least give me some details on how to reproduce the settings on your Azure Blob storage so I can have a look at what is going on.

**Comment by EdwardMcFlurry** [comment]
I’ll give you few links of the videos I tried with so far, including those uploaded as samples on the internet and those uploaded alongside my project as I wanted to understand if the top-level domains contribute to this ISSUE but also those uploaded on Azure.

Please start to test this ISSUE by cloning this project and then test with your own implementation of Media View/Element.

Can I email you those links for sake of the privacy of what I’m working on 🙏🏻

**Comment by jfversluis** [comment]
Sure you can send it to the email address that you can find on my GitHub profile.

In addition, you might want to check this blog post out which describes the same: https://blog.pieeatingninjas.be/2021/12/21/fixing-video-file-not-streaming-from-azure-blob-storage-on-android/

Seems configuration on the Azure side is causing this. I will see if there is anything I can do on our side to make it better, but at least you should be unblocked by this.

**Comment by EdwardMcFlurry** [comment]
@jfversluis Email sent, I did take a look at the Azure Portal hoping to find something there to tweak but nothing and also with the code on MVC project, I'm only uploading the Blobs in this case the videos but don't know how to configure this option from the blog post.

Do you perhaps have an idea as to where can I tweak it in code or in portal?

I'm just wondering in terms of the blog post you referenced that the Videos cannot play on this "plugin" on Android but did play on your project as far as something needs to be tweaked for Azure Blob Storage 🤷🏻‍♂️

**Comment by EdwardMcFlurry** [comment]
Hello @davidbritch, thank you so much for your efforts with the initial video player. I remember last year searching through Google on Video Player in Maui but nothing was available at that time and weeks later boom I bumped into your blog post where you explained neatly on how to have Videos playing in Maui.

I cloned the project and went through line by line trying to figure it out and eventually got it working in my project. You came back with changes made by Apple to iOS 16 regarding the UI controls for the video player and again I appreciated all that effort but 😓 you disappeared just like that into a thin air and initially I thought like you were getting married or something 🤣 but overall thank you 🙏🏻 

Then @jfversluis came along 🕺I followed MediaElement from day one in CommunityToolkit repo as PR with my fingers crossed that you’ll finish all the Amazing work on it and today I’m happy to say that I integrated in my project/app SUCCESSFULLY the **MediaElement** 2 hours after it dropped on Nuget and it works like a charm 👍🏻 on Android with all the videos on Azure Blob Storage recorded with iPhone, iPad and typical Smartphone.

I like the Buffering functionality with MediaOpened/Failed/Ended event handlers and damn you dit it.

Thank you so much for an incredible work all of you do on Maui 😎 I’ll be closing this issue soon.
